up42/utils.py

```diff
@@ -71,18 +72,17 @@ def download_results_from_gcs(
                 f"Connection error, please try again! {err}"
             )
 
-    # Unpack and exclude data.json
-    out_filepaths: List[str] = []
     with tarfile.open(tgz_file) as tar:
-        members = tar.getmembers()
-        files = [f for f in members if f.isfile()]
-        for file in files:
-            f = tar.extractfile(file)
-            content = f.read()  # type: ignore
-            out_fp = output_directory / f"{Path(file.name).name}"
-            with open(out_fp, "wb") as dst:
-                dst.write(content)
-            out_filepaths.append(str(out_fp))
+        tar.extractall(path=output_directory)
+        output_folder_path = output_directory / "output"
+        for src_path in output_directory.glob("output/**/*"):
```

Comment:
So the try except block is there because for some reason, globbing the output folder returns the output directory in this list - this just make sure all the files are moved (I checked this again and removed it, seems the error I got before is gone). The solution you propose would not work since you actually have to move files and folders around, not just rename them/change the paths. Also the solution in SO we would come into the same problem of not keeping the folder structure.

Reply:
👍

Reply:
Not sure if deliberatly globbing `output_directory.glob("output/**/*")`, but maybe the previous issue with the directory would be even saver with globbing in the folder directly instead of using output in the pattern: `output_folder_path.rglob("*")`

Reply:
Good point, will change.